backend/src/main/java/com/benine/backend/database/MySQLDatabase.java

```diff
@@ -232,6 +220,97 @@ public void addCamera(int id, String macAddress) {
   }
 
   @Override
+  public void checkCameras() throws SQLException {
+    ArrayList<Camera> cameras = ServerController.getInstance().getCameraController().getCameras();
+    ArrayList<String> macs = new ArrayList<String>();
+    ResultSet resultset = null;
+    Statement statement = null;
+    try {
+      statement = connection.createStatement();
+      String sql = "SELECT ID, MACAddress FROM camera";
+      resultset = statement.executeQuery(sql);
+      statement.close();
+      resultset.close();
+      checkOldCameras(resultset, cameras, macs);
+      checkNewCameras(cameras, macs);
+    } catch (SQLException e) {
+      e.printStackTrace();
+      getLogger().log("Cameras could not be gotten from database.", LogEvent.Type.CRITICAL);
+    } catch (CameraConnectionException e) {
+      e.printStackTrace();
+    } finally {
+      if (statement != null) {
+        statement.close();
+        resultset.close();
+      }
+    }
+  }
+
+  /**
+   * Cheks if there are cameras in the database to be deleted.
+   * @param result The resultset from the query
+   * @param cameras The cameras
+   * @param macs The MACAddresses of the cameras in the database
+   * @throws SQLException No right connection to the database
+   * @throws CameraConnectionException Not able to connect to the camera
+   */
+  public void checkOldCameras(ResultSet result, ArrayList<Camera> cameras, ArrayList<String> macs)
+      throws SQLException, CameraConnectionException {
+    while (result.next()) {
```

Comment:
I think it would be better to have a seperate method to retrieve the mac address from the result set.
Then you can just pass only the mac address and the camera's to this method.